### ISSUE TITLE
Trouble connecting to CODE (Collabora) docker container with Nextcloud

### ISSUE THREAD

**Issue by xthursdayx** [nextcloud/docker#809]
Hi, I’m running Nextcloud and Collabora as docker containers on my unRAID server. Both are accessible via a LetsEncrypt reverse proxy. I also have the CODE app installed within Nextcloud and the settings updated. Nextcloud generally works great, but whenever I try to open a document it fails and I get the message “Failed to connect to Collabora Online Development Edition. Please try again later or contact your server administrator.”

Both Collabora’s general “OK” interface and the admin panel are accessible at https://collabora.mywebsite.com and this is the setting that is set in Nextcloud's CODE setting. I’ve been trying to get this to work for months, but haven’t had any luck. Nextcloud’s logs say nothing, while Collabora’s system logs do have some errors, however, I’m not sure if they are relevant. I copied them below just in case. If anybody can help me to figure this out I’d be super grateful. Thanks.

A selection of the most common errors:

```
wsd-00031-00906 2019-07-12 06:12:14.153253 [ docbroker_043 ] WRN Client session [0043] not found to forward message: o1571 statechanged: .uno:AssignLayout=| wsd/DocumentBroker.cpp:1798

wsd-00031-00032 2019-07-12 06:12:14.904970 [ prisoner_poll ] WRN Prisoner connection disconnected but without valid socket.| wsd/LOOLWSD.cpp:1783
wsd-00031-00940 2019-07-12 06:12:17.593687 [ docbroker_048 ] ERR Socket #57 SSL error: SYSCALL (5). (EPIPE: Broken pipe)| ./net/SslSocket.hpp:236
wsd-00031-00940 2019-07-12 06:12:17.593765 [ docbroker_048 ] ERR #57: Wrote outgoing data -1 bytes. (EPIPE: Broken pipe)| ./net/Socket.hpp:1058

wsd-00031-00032 2019-07-12 06:12:23.378568 [ prisoner_poll ] WRN ForKit not responsive for 7046 ms forking 1 children. Resetting.| wsd/LOOLWSD.cpp:427
wsd-00031-01162 2019-07-12 06:12:23.581930 [ docbroker_056 ] WRN getNewChild: No child available. Sending spawn request to forkit and failing.| wsd/LOOLWSD.cpp:552
```

**Comment by xthursdayx** [comment]
Also, I'm not sure if this is the right repository to post this on, but I have been able to find help elsewhere.

**Comment by J0WI** [comment]
I assume the errors come from the collabora conatiner?
It's hard to help without seeing any configuration details. You may find a helpful example in #630.

**Comment by blackerking** [comment]
You may have to change the portnumber of nextcloud to standard (80 HTTP ; 443 HTTPS)

**Comment by xthursdayx** [comment]
Sorry for the delay. The errors I listed are indeed coming from the Collabora container. 

Here is my configuration, and I've also attached my docker container configurations for Nextcloud and Collabora and the settings for CODE from within Nextcloud:
```
<?php
$CONFIG = array (
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'filelocking.enabled' => 'true',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' =>
  array (
    'host' => '192.168.1.107',
    'port' => 6379,
    'timeout' => 0.0,
  ),
  'datadirectory' => '/data',
  'instanceid' => 'ocftakwz6q8m',
  'passwordsalt' => 'Ev1afHP5nkkdUnL78XAA7ltNREucPb',
  'secret' => 'yFxiE8RIyd5CMcq5x2DLuELEcWL1VoreaAjkFo+qPYj4OR/6',
  'trusted_domains' =>
  array (
    0 => '192.168.1.107:444',
    1 => 'nextcloud.my.website',
    2 => 'collabora.my.website',
  ),
  'trusted_proxies' =>
  array (
    0 => 'letsencrypt',
  ),
  'overwrite.cli.url' => 'https://nextcloud.my.website',
  'overwritehost' => 'nextcloud.my.website',
  'overwrite    protocol' => 'https',
  'dbtype' => 'mysql',
  'version' => '16.0.3.0',
  'dbname' => 'nextcloud',
  'dbhost' => '192.168.1.107:3306',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'mysql.utf8mb4' => true,
  'dbuser' => 'nextcloud',
  'dbpassword' => 'xxxxxxxxxxxxxx',
  'installed' => true,
  'mail_from_address' => 'my_email',
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_domain' => 'gmail.com',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'smtp.gmail.com',
  'mail_smtpport' => '465',
  'mail_smtpsecure' => 'ssl',
  'mail_smtpname' => 'my_email@gmail.com',
  'mail_smtppassword' => 'xxxxxxxxxx',
  'maintenance' => false,
  'theme' => '',
  'loglevel' => 0,
  'twofactor_enforced' => 'true',
  'twofactor_enforced_groups' =>
  array (
  ),
  'twofactor_enforced_excluded_groups' =>
  array (
  ),
  'ldapIgnoreNamingRules' => false,
  'ldapProviderFactory' => 'OCA\\User_LDAP\\LDAPProviderFactory',
);
```
<img width="1059" alt="Screen Shot 2019-07-19 at 12 47 39" src="https://user-images.githubusercontent.com/18044308/61551647-3cbd1200-aa45-11e9-8560-d58c3e97cf91.png">
<img width="1011" alt="Screen Shot 2019-07-19 at 12 31 24" src="https://user-images.githubusercontent.com/18044308/61551648-3d55a880-aa45-11e9-8677-65029325e382.png">
<img width="1019" alt="Screen Shot 2019-07-19 at 12 30 47" src="https://user-images.githubusercontent.com/18044308/61551650-3d55a880-aa45-11e9-8f3a-a2f7cf2d8359.png">

**Comment by xthursdayx** [comment]
> You may have to change the portnumber of nextcloud to standard (80 HTTP ; 443 HTTPS)

The OS my server runs (unRAID) uses port 443 for its interface, so I've changed the default port to 444.

**Comment by xthursdayx** [comment]
I'm also getting these errors from Collabora:

```
wsd-00030-00039 2019-07-19 17:18:33.190587 [ websrv_poll ] ERR Socket #21 SSL BIO error: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca (0: Success)| ./net/SslSocket.hpp:281
wsd-00030-00039 2019-07-19 17:18:33.190972 [ websrv_poll ] ERR Error while handling poll for socket #21 in websrv_poll: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca| ./net/Socket.hpp:570
wsd-00030-00039 2019-07-19 17:18:36.301993 [ websrv_poll ] WRN client - server version mismatch, disabling browser cache.| wsd/FileServer.cpp:279
wsd-00030-00039 2019-07-19 17:18:36.509187 [ websrv_poll ] ERR FileServerRequestHandler::NotAuthenticated: No authentication information found| 
```

**Comment by pelaxa** [comment]
Were you able to resolve your issue?  I also have a similar setup and both Nextcloud and Collabora seems to be working fine individually (separate containers), but I cannot get them to talk properly.

Do the containers have to have a shared volume or anything?

My error is a little different though:
`{"reqId":"c6Jsk77JfFdbcFS8ymoj","level":3,"time":"2019-09-29T17:18:14+00:00","remoteAddr":"172.20.0.2","user":"XXXXX","app":"index","method":"GET","url":"\/apps\/richdocuments\/index?fileId=28372&requesttoken=Bx0uEt8G2u9Xe7rBVXu%2FHXv0TJXq4822f44Gs%2F5d9cc%3D%3AMXxvQrozrIY2H%2FyWIBP8TwilY9%2BgmfnTHc9x4rATnfM%3D","message":{"Exception":"OCP\\AutoloadNotAllowedException","Message":"Autoload path not allowed: \/var\/www\/html\/apps\/federation\/lib\/trustedservers.php","Code":0,"Trace":[{"file":"\/var\/www\/html\/lib\/autoloader.php","line":164,"function":"isValidPath","class":"OC\\Autoloader","type":"->","args":["\/var\/www\/html\/apps\/federation\/lib\/trustedservers.php"]},{"function":"load","class":"OC\\Autoloader","type":"->","args":["OCA\\Federation\\TrustedServers"]},{"function":"spl_autoload_call","args":["OCA\\Federation\\TrustedServers"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":96,"function":"__construct","class":"ReflectionClass","type":"->","args":["OCA\\Federation\\TrustedServers"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":119,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Federation\\TrustedServers"]},{"file":"\/var\/www\/html\/lib\/private\/ServerContainer.php","line":150,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Federation\\TrustedServers"]},{"file":"\/var\/www\/html\/apps\/richdocuments\/lib\/Service\/FederationService.php","line":59,"function":"query","class":"OC\\ServerContainer","type":"->","args":["OCA\\Federation\\TrustedServers"]},{"function":"__construct","class":"OCA\\Richdocuments\\Service\\FederationService","type":"->","args":[{"__class__":"OC\\Memcache\\Factory"},{"__class__":"OC\\Http\\Client\\ClientService"},{"__class__":"OC\\AppFramework\\Logger"},{"__class__":"OCA\\Richdocuments\\TokenManager"}]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":81,"function":"newInstanceArgs","class":"ReflectionClass","type":"->","args":[[{"__class__":"OC\\Memcache\\Factory"},{"__class__":"OC\\Http\\Client\\ClientService"},{"__class__":"OC\\AppFramework\\Logger"},{"__class__":"OCA\\Richdocuments\\TokenManager"}]]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":98,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":[{"name":"OCA\\Richdocuments\\Service\\FederationService","__class__":"ReflectionClass"}]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":119,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Richdocuments\\Service\\FederationService"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":411,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Richdocuments\\Service\\FederationService"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":382,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Richdocuments\\Service\\FederationService"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":68,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Richdocuments\\Service\\FederationService"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":98,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":[{"name":"OCA\\Richdocuments\\Controller\\DocumentController","__class__":"ReflectionClass"}]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":119,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Richdocuments\\Controller\\DocumentController"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":411,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Richdocuments\\Controller\\DocumentController"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":382,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Richdocuments\\Controller\\DocumentController"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/App.php","line":113,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Richdocuments\\Controller\\DocumentController"]},{"file":"\/var\/www\/html\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Richdocuments\\Controller\\DocumentController","index",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"richdocuments.document.index"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"richdocuments.document.index"}]},{"file":"\/var\/www\/html\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"richdocuments.document.index"}]},{"file":"\/var\/www\/html\/lib\/base.php","line":975,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/richdocuments\/index"]},{"file":"\/var\/www\/html\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/var\/www\/html\/lib\/autoloader.php","Line":139,"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit\/605.1.15 (KHTML, like Gecko) Version\/12.1.2 Safari\/605.1.15","version":"16.0.4.1"}`

**Comment by blackerking** [comment]
I've seen some similiar errors.
Solution:
Open Nextcloud via browser and open the public IP(not the Docker IP).
Configure Nextcloud to use Collabora(or LOOL) with the same public IP and the correct port.

Such configuration must work - everything else makes errors.

**Comment by pelaxa** [comment]
That's interesting.  So I have this setup using domain names, and internally vs externally they resolve to different IP addresses.  Do you think that is the issue here?

**Comment by blackerking** [comment]
> so is this a bug or somehow still configuration related?

I don't know exactly but maybe my thoughts can help you.

I've testet NC with LOOL and Collabora.
After testing many things I came to this conclusion:

* Always open NC and LOOL/Collabora with the extern IP(or Hostname).
-> So you never got to add extra hostnames into NC
* For testing LOOL or Collabora try to open: hostname:9980/loleaflet/dist/admin/admin.html
* Don't forget to add the LOOL or Collabora URL inside the App of NC
-> Everytime I've changed the version, I had to add the URL again!

Maybe the start script needs to run another time. With this environment argument:

```
  libreoffice:

    environment:
      domain: nextcloud

```
